### PR TITLE
fix: ensure REST API CFN outputs the API ID

### PR DESCRIPTION
**Pull request by cjihrig**
#### Description of changes
This commit updates the REST API migration logic to ensure that the API ID is a CFN Output, as the policy consolidation logic will rely on this during push.

#### Description of how you validated changes
Manual testing

#### Checklist
- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ammarkarachi — APPROVED** [review]

File: packages/amplify-provider-awscloudformation/src/utils/consolidate-apigw-policies.ts [hunk 1]
```diff
@@ -246,8 +246,14 @@ function updateExistingApiCfn(context: $TSContext, api: $TSObject): void {
   const apiParamsFile = path.join(resourceDir, API_PARAMS_FILE);
   const cfn: any = JSONUtilities.readJson(cfnTemplate, { throwIfNotExist: false }) ?? {};
   const parameterJson = JSONUtilities.readJson(paramsFile, { throwIfNotExist: false }) ?? {};
-  const parameters = cfn?.Parameters ?? {};
-  const resources = cfn?.Resources ?? {};
+
+  if (!cfn) {
+    // This should not happen. If it does, nothing else will work anyway so just bail out.
```
Comment (attilah) [inline packages/amplify-provider-awscloudformation/src/utils/consolidate-apigw-policies.ts:251]: Can't we just throw an error to make sure a return here will not makes things bad? We have a similar throw for layers when a value MUST be present that further functionality depends on.

**Review by attilah — APPROVED** [review]
LGTM a nit about the return

File: packages/amplify-provider-awscloudformation/src/utils/consolidate-apigw-policies.ts (hunk 1 above)
Comment (cjihrig) [reply packages/amplify-provider-awscloudformation/src/utils/consolidate-apigw-policies.ts:251]: Sure, I can update it to throw.

Comment (cjihrig) [reply packages/amplify-provider-awscloudformation/src/utils/consolidate-apigw-policies.ts]: Updated!

Comment (ammarkarachi) [reply packages/amplify-provider-awscloudformation/src/utils/consolidate-apigw-policies.ts]: I think throwing returning would make the CLI proceed with pushing the category

**Pull request merged**